src/lib/TwaManifest.ts

```diff
@@ -61,34 +64,19 @@ function generatePackageId(host: string): string {
 /**
  * A wrapper around the WebManifest's ShortcutInfo.
  */
-class ShortcutInfo {
-  name: string;
-  shortName: string;
-  url: string | undefined;
-  chosenIconUrl: string | undefined;
-
+export class ShortcutInfo {
   /**
-   * @param {Object} the WebManifest's ShortcutInfo.
-   * @param {URL} webManifestUrl the URL where the webmanifest is available.
+   * @param {string} name
+   * @param {string} shortName
+   * @param {string} url target Url for when the shortcut is clicked
+   * @param {string} chosenIconUrl Url for the icon
    */
-  constructor(shortcutInfo: WebManifestShortcutJson, webManifestUrl: URL) {
-    this.name = shortcutInfo['name'] || '';
-    this.shortName = shortcutInfo['short_name'] || this.name;
-    this.url = shortcutInfo['url'] ?
-      new URL(shortcutInfo['url'], webManifestUrl).toString() : undefined;
-
-    if (shortcutInfo.icons === undefined) {
-      return;
-    }
-
-    const suitableIcon = util.findSuitableIcon(shortcutInfo.icons, 'any');
-    if (suitableIcon !== null) {
-      this.chosenIconUrl = new URL(suitableIcon.src, webManifestUrl).toString();
-    }
-  }
-
-  isValid(): boolean {
-    return this.name != undefined && this.url != undefined && this.chosenIconUrl != undefined;
+  constructor(readonly name: string, readonly shortName: string, readonly url: string,
+        readonly chosenIconUrl: string) {
+    this.name = name;
```

Comment:
You're using [Parameter Properties][1] here right? I don't think you need lines 76 to 79.

[1]: https://www.typescriptlang.org/docs/handbook/classes.html#parameter-properties

Reply:
This is great! Updated!